src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/DefaultGeneratedDocumentPublisher.cs

```diff
@@ -78,13 +78,14 @@ public override void PublishCSharp(string filePath, SourceText sourceText, long
             {
                 HostDocumentFilePath = filePath,
                 Changes = textChanges,
-                HostDocumentVersion = hostDocumentVersion,
+                HostDocumentVersion = (long)hostDocumentVersion,
             };
 
-            _server.Value.Client.SendRequest(LanguageServerConstants.RazorUpdateCSharpBufferEndpoint, request);
+            var result = _server.Value.Client.SendRequest(LanguageServerConstants.RazorUpdateCSharpBufferEndpoint, request);
+            result.ReturningVoid(System.Threading.CancellationToken.None);
```

Comment:
Can you explain what this does?

Reply:
This is the call that actually makes the request, any SendRequest without a .Returning* after it will do nothing. SendRequest now just constructs the request.

Reply:
oh interesting. That seems like a funky pattern @david-driscoll 😄. Why not fire the request off in the background and have the "ReturningXYZ" be just the awaiting portion of the pattern?